### PR TITLE
Hide preview button for 'EscapedMarkupFormatter'

### PR DESCRIPTION
**Pull request by janfaracik**
Unlike other formatters, the 'Preview' link doesn't do anything for the default 'EscapedMarkupFormatter' formatter, clicking it just presents back the escaped version of the text the user has entered in. Intention would be to remove this link for the default formatter where afaik it doesn't serve a purpose, and keep it for other formatters.

**Before**
image

**After**
image

### Testing done

* Preview link isn't visible for 'EscapedMarkupFormatter'
* Preview link is visible for other formatters

### Proposed changelog entries

- Hide preview button for 'EscapedMarkupFormatter'

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@jenkinsci/sig-ux 

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by daniel-beck** [comment]
> clicking it just presents back the escaped version of the text the user has entered in

Mostly at least. A notable exception is that tab characters aren't retained. (Notably, consecutive spaces and line breaks are both retained.)

I'm ±0 on this. It looks a little nicer but loses some information (more about which formatter it is than what it looks like).

**Comment by MarkEWaite** [comment]
I'm "+1" for the change but I think it will need to be checked that it does not break tests in the plugin bill of materials or in the acceptance test harness.

File: core/src/main/resources/lib/form/textarea.jelly [hunk 1]
```diff
@@ -105,7 +105,8 @@ THE SOFTWARE.
       <j:mute>${customizedFields.add(name)}</j:mute>
     </j:if>
   </f:possibleReadOnlyField>
-  <j:if test="${attrs.previewEndpoint!=null and !readOnlyMode}">
+
+  <j:if test="${attrs.previewEndpoint!=null and !readOnlyMode and app.markupFormatter.class.name != 'hudson.markup.EscapedMarkupFormatter'}">
```
Comment (jtnord) [inline core/src/main/resources/lib/form/textarea.jelly:109]: Am I missing something?
Nothing ties previews to markup formatters does it?  if so nothing here says the preview needs to be done using only a the Jenkins `MarkupFormatter`. 

```suggestion
  <j:if test="${attrs.previewEndpoint!=null and !readOnlyMode and (app.markupFormatter.class.name != 'hudson.markup.EscapedMarkupFormatter or attrs.previewEndpoint != '/markupFormatter/previewDescription')'}">
```

That is I could have a jenkins using the `EscapedMarkupFormatter` but I may have set the previewEndPoint to `MyDescriptorImpl.preview(String)`  

Indeed, the test on line 111 does imply that I should be able to use this irrespective of any global markup formatter.

a completely contrived example could be where a user supplies some toml, and my descriptor takes that toml (per job) and merges it with some global toml to create the "end toml" which is then displayed to the user?

**Review by jtnord — COMMENTED** [review]
code logic looks incorrect in that it ignores any user defined preview endpoint

**Comment by timja** [comment]
@janfaracik did you want to continue with this PR?

**Comment by janfaracik** [comment]
I'll come back to this in the future 👍  Thanks all

**Pull request closed without merging**